spec/features/saml/saml_spec.rb

```diff
@@ -167,6 +167,18 @@
   end
 
   context 'visiting /api/saml/logout' do
+    context 'via SP-initiated logout when not logged in to IdP' do
+      it 'does not raise an exception' do
+        request = OneLogin::RubySaml::Logoutrequest.new
+        settings = sp1_saml_settings
+        settings.name_identifier_value = SecureRandom.uuid
+
+        expect do
+          visit request.create(settings)
+        end.to_not raise_error
+      end
+    end
```

Comment:
This is not a valid spec. The bug we are purporting to fix relates to SLO and session state at IdP. We need to test the following:

1. A user authenticates with a valid SP
2. The IdP session times out. `Timecop.travel` more than `Devise.timeout_in`
3. The SLO request from SP is properly handled by issuing a SAMLResponse with Success status and verification that the IdP session is no longer active.